### PR TITLE
Build a side panel component and integrate into dashboard's context menu

### PR DESCRIPTION
**Pull request by nrwinner**
This PR introduces a reusable side-panel component and lays the groundwork for building the revision pipeline in the client. The side panel is feature-toggled currently and will only be accessible in the development environment.

Screen Shot 2019-06-19 at 3 48 53 PM

### CODE REVIEW

**Comment by nrwinner** [comment]
[ch27]

File: src/app/onion/dashboard/dashboard.component.ts [hunk 1]
```diff
@@ -791,6 +792,16 @@ export class DashboardComponent implements OnInit, OnDestroy {
     this.changelogs = undefined;
   }
 
+   /**
+   * Opens the Change Log modal for a specified Learning Object and fetches the appropriate change logs
```
Comment (DonSeannelly) [inline src/app/onion/dashboard/dashboard.component.ts]: I'm a little confused by this `sidePanelWatcher`. Is this JSDoc perhaps a copy and paste error - or am I missing something?

Is the watcher observing the side panel, or actively controlling it?

Comment (nrwinner) [reply src/app/onion/dashboard/dashboard.component.ts]: That's certainly a copy and paste error.

The watcher both watches and controls. It does allow for direct manipulation interaction of the side panel (opening and closing from the host component) but it also allows the host component to monitor close events originating from within the side panel. I'm not attached to the name so if it's confusing we should change it.

File: src/app/shared/side-panel/side-panel-viewer/side-panel-viewer.component.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    </ng-container>
+  `,
+  styleUrls: ['./side-panel-viewer.component.scss'],
+  animations: [slide, fade]
+})
+export class SidePanelViewerComponent implements OnInit, OnDestroy {
+  _watcher$: BehaviorSubject<boolean>;
+  contentWidth = 400;
+
+  value: boolean;
```
[the remark this answers is not in the document]
Comment (nrwinner) [reply src/app/shared/side-panel/side-panel-viewer/side-panel-viewer.component.ts]: Value is indicative of whether or not the panel is open or closed (true if open, false if closed). That can be made more descriptive if it's confusing.

[the remark this answers is not in the document]
Comment (DonSeannelly) [reply src/app/shared/side-panel/side-panel-viewer/side-panel-viewer.component.ts]: I would be in favor of going with a specific name. Maybe `isOpen`

**Pull request merged**